app/views/ChooseProvider.js

```diff
@@ -151,30 +152,19 @@ class ChooseProviderScreen extends Component {
     );
   }
 
-  setUrlText(urlText) {
-    this.setState({ urlText });
-  }
-
-  /**
-   * Checks if the user selected any authorities whose `url` matches
-   * the `url` param.
-   * @param {string} url
-   */
-  hasExistingAuthorityWithUrl(url) {
-    return this.state.selectedAuthorities.some(x => x.url === url);
-  }
+  setUrlText = urlText => this.setState({ urlText });
 
   /**
    * Reset the URL input field to it's original/default settings
    */
-  resetUrlInput() {
+  resetUrlInput = () =>
     this.setState({
       displayUrlEntry: 'none',
       urlEntryInProgress: false,
+      urlText: '',
     });
-  }
 
-  addCustomUrlToState() {
+  addCustomUrlToState = () => {
     let { urlText: url, selectedAuthorities } = this.state;
 
     if (!validUrl.isWebUri(url)) {
```

Comment:
So this is actually my fault - this should be 

```js
if (!validUrl.isWebUri(url) || this.hasExistingAuthorityWithUrl(url)) {
```

Looks like I accidentally removed that check when I included the `validUrl` library.